pkg/provider/aws/action/eks/eks.go

```diff
@@ -0,0 +1,268 @@
+		AssumeRolePolicy: pulumi.String(`{
+			"Version": "2012-10-17",
+			"Statement": [{
+					"Sid": "",
+					"Effect": "Allow",
+					"Principal": {
+							"Service": "ec2.amazonaws.com"
+					},
+					"Action": "sts:AssumeRole"
+			}]
+	}`),
+	})
+	if err != nil {
+		return err
+	}
+	nodeGroupPolicies := []string{
+		"arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+		"arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+		"arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+	}
+	for i, nodeGroupPolicy := range nodeGroupPolicies {
+		_, err := iam.NewRolePolicyAttachment(ctx, fmt.Sprintf("ngpa-%d", i), &iam.RolePolicyAttachmentArgs{
+			Role:      nodeGroupRole.Name,
+			PolicyArn: pulumi.String(nodeGroupPolicy),
+		})
+		if err != nil {
+			return err
+		}
+	}
+	// Create a Security Group that we can use to actually connect to our cluster
+	clusterSg, err := ec2.NewSecurityGroup(ctx, "cluster-sg", &ec2.SecurityGroupArgs{
+		VpcId: pulumi.String(vpc.Id),
+		Egress: ec2.SecurityGroupEgressArray{
+			ec2.SecurityGroupEgressArgs{
+				Protocol:   pulumi.String("-1"),
+				FromPort:   pulumi.Int(0),
+				ToPort:     pulumi.Int(0),
+				CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+			},
+		},
+		Ingress: ec2.SecurityGroupIngressArray{
+			ec2.SecurityGroupIngressArgs{
+				Protocol:   pulumi.String("tcp"),
+				FromPort:   pulumi.Int(80),
+				ToPort:     pulumi.Int(80),
+				CidrBlocks: pulumi.StringArray{pulumi.String("0.0.0.0/0")},
+			},
+		},
+	})
+	if err != nil {
+		return err
+	}
+	// Create EKS Cluster
+	eksCluster, err := eks.NewCluster(ctx, "eks-cluster", &eks.ClusterArgs{
+		RoleArn: pulumi.StringInput(eksRole.Arn),
+		VpcConfig: &eks.ClusterVpcConfigArgs{
+			PublicAccessCidrs: pulumi.StringArray{
+				pulumi.String("0.0.0.0/0"),
+			},
+			SecurityGroupIds: pulumi.StringArray{
+				clusterSg.ID().ToStringOutput(),
+			},
+			SubnetIds: toPulumiStringArray(subnet.Ids),
+		},
+		Version: pulumi.String(r.KubernetesVersion),
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, addon := range r.Addons {
+		_, err = eks.NewAddon(ctx, addon, &eks.AddonArgs{
+			ClusterName: eksCluster.Name,
+			AddonName:   pulumi.String(addon),
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	_, err = eks.NewNodeGroup(ctx, "node-group-0", &eks.NodeGroupArgs{
+		ClusterName:   eksCluster.Name,
+		NodeGroupName: pulumi.String("eks-nodegroup-0"),
+		NodeRoleArn:   pulumi.StringInput(nodeGroupRole.Arn),
+		SubnetIds:     toPulumiStringArray(subnet.Ids),
+		InstanceTypes: pulumi.StringArray{
+			pulumi.String(r.VMSize),
+		},
+		ScalingConfig: &eks.NodeGroupScalingConfigArgs{
+			DesiredSize: pulumi.Int(r.ScalingDesiredSize),
+			MaxSize:     pulumi.Int(r.ScalingMaxSize),
+			MinSize:     pulumi.Int(r.ScalingMinSize),
+		},
+	})
+	if err != nil {
+		return err
+	}
+
+	kubeconfig := generateKubeconfig(eksCluster.Endpoint, eksCluster.CertificateAuthority.Data().Elem(), eksCluster.Name)
```

Comment:
:thinking: is this working? I assume you would need something like this:

https://www.pulumi.com/docs/iac/concepts/inputs-outputs/all/#creating-a-new-data-structure

Reply:
:+1: